shell/platform/android/io/flutter/plugin/editing/SpellCheckPlugin.java

```diff
@@ -0,0 +1,135 @@
+  }
+
+  /**
+   * Unregisters this {@code SpellCheckPlugin} as the {@code
+   * SpellCheckChannel.SpellCheckMethodHandler}, for the {@link
+   * io.flutter.embedding.engine.systemchannels.SpellCheckChannel}, and closes the most recently
+   * opened {@code SpellCheckerSession}.
+   *
+   * <p>Do not invoke any methods on a {@code SpellCheckPlugin} after invoking this method.
+   */
+  public void destroy() {
+    mSpellCheckChannel.setSpellCheckMethodHandler(null);
+
+    if (mSpellCheckerSession != null) {
+      mSpellCheckerSession.close();
+    }
+  }
+
+  @Override
+  public void initiateSpellCheck(@NonNull String locale, @NonNull String text) {
+    performSpellCheck(locale, text);
+  }
+
+  /** Calls on the Android spell check API to spell check specified text. */
+  public void performSpellCheck(String locale, String text) {
+    String[] localeCodes = locale.split("-");
+    Locale localeFromString = LocalizationPlugin.localeFromString(locale);
+
+    if (mSpellCheckerSession != null) {
+      mSpellCheckerSession.close();
+    }
+    mSpellCheckerSession =
+        mTextServicesManager.newSpellCheckerSession(
+            null,
+            localeFromString,
+            this,
+            /** referToSpellCheckerLanguageSettings= */
+            true);
+
+    TextInfo[] textInfos = new TextInfo[] {new TextInfo(text)};
+    mSpellCheckerSession.getSentenceSuggestions(textInfos, MAX_SPELL_CHECK_SUGGESTIONS);
+  }
+
+  /**
+   * Callback for Android spell check API that decomposes results and send results through the
+   * {@link SpellCheckChannel}.
+   *
+   * <p>Spell check results will be encoded as a string representing the span of that result, with
+   * the format [start_index.end_index.suggestion_1,suggestion_2,suggestion_3], where there may be
+   * up to 5 suggestions.
+   */
+  @Override
+  public void onGetSentenceSuggestions(SentenceSuggestionsInfo[] results) {
+    ArrayList<String> spellCheckerSuggestionSpans = new ArrayList<String>();
+
+    if (results.length > 0) {
+      SentenceSuggestionsInfo spellCheckResults = results[0];
+
+      for (int i = 0; i < spellCheckResults.getSuggestionsCount(); i++) {
+
+        SuggestionsInfo suggestionsInfo = spellCheckResults.getSuggestionsInfoAt(i);
+        int suggestionsCount = suggestionsInfo.getSuggestionsCount();
+
+        if (suggestionsCount > 0) {
+          String spellCheckerSuggestionSpan = "";
+          int start = spellCheckResults.getOffsetAt(i);
+          int length = spellCheckResults.getLengthAt(i);
+
+          spellCheckerSuggestionSpan += (String.valueOf(start) + ".");
+          spellCheckerSuggestionSpan += (String.valueOf(start + (length - 1)) + ".");
+
+          for (int j = 0; j < suggestionsCount; j++) {
+            spellCheckerSuggestionSpan += (suggestionsInfo.getSuggestionAt(j) + ",");
+          }
+
+          spellCheckerSuggestionSpans.add(
+              spellCheckerSuggestionSpan.substring(0, spellCheckerSuggestionSpan.length() - 1));
```

Comment:
is `spellCheckerSuggestionSpan.substring(0, spellCheckerSuggestionSpan.length() - 1)` just `spellCheckerSuggestionSpan`?

Reply:
No, I'm chopping off the last separator of the suggested replacements list.